lib/hooks/userspace/hook-mysql.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var cls = require('../../../lib/cls.js');
+var shimmer = require('shimmer');
+var semver = require('semver');
+var agent;
+
+var SUPPORTED_VERSIONS = '~2.9.x';
+
+function createQueryWrap(createQuery) {
+  return function createQuery_trace(sql, values, cb) {
+    var root = cls.getRootContext();
+    if (!root) {
+      agent.logger.warn('Cannot trace mysql outside of a supported framework.');
+      return createQuery.apply(this, arguments);
+    }
+    var span = agent.startSpan('mysql-query');
+    var query = createQuery.apply(this, arguments);
+    span.addLabel('sql', query.sql);
+    cls.getNamespace().bindEmitter(query);
```

Comment:
We need to do this to maintain context. Even if we don't provide a callback, the query object has internal only event emitters attached to it to manage connection related lifecycle. See https://github.com/felixge/node-mysql/blob/master/lib/protocol/Protocol.js#L142